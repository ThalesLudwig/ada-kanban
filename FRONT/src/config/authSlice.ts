import { createSlice } from "@reduxjs/toolkit";

export type AuthProps = {
  token: string;
};

const initialState: AuthProps = {
  token: "",
};

const authSlice = createSlice({
  name: "theme",
  initialState,
  reducers: {
    setToken(state, action) {
      state.token = action.payload;
    },
  },
});

export const { setToken } = authSlice.actions;

export default authSlice.reducer;
