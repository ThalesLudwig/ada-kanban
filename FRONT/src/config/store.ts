import storage from "redux-persist/lib/storage";
import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import { persistStore, persistReducer } from "redux-persist";

import themeReducer from "./themeSlice";
import authReducer from "./authSlice";

export const rootReducer = combineReducers({
  theme: persistReducer({ key: "theme", storage }, themeReducer),
  auth: persistReducer({ key: "auth", storage }, authReducer),
});

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export const persistor = persistStore(store);

export type AppDispatch = typeof store.dispatch;

export type RootState = ReturnType<typeof store.getState>;

export default store;
