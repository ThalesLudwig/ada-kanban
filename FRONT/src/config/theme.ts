const light = {
  primary: "rgb(0, 98, 158)",
  onPrimary: "rgb(255, 255, 255)",
  primaryContainer: "rgb(207, 228, 255)",
  onPrimaryContainer: "rgb(0, 29, 52)",
  error: "rgb(186, 26, 26)",
  onError: "rgb(255, 255, 255)",
  errorContainer: "rgb(255, 218, 214)",
  onErrorContainer: "rgb(65, 0, 2)",
  background: "rgb(252, 252, 255)",
  onBackground: "rgb(26, 28, 30)",
  surface: "rgb(222, 227, 235)",
  onSurface: "rgb(66, 71, 78)",
  outline: "rgb(115, 119, 127)",
  elevation: {
    level0: "transparent",
    level1: "rgb(239, 244, 250)",
  },
  surfaceDisabled: "rgba(26, 28, 30, 0.12)",
  onSurfaceDisabled: "rgba(26, 28, 30, 0.38)",
};

const dark = {
  primary: "rgb(154, 203, 255)",
  onPrimary: "rgb(0, 51, 85)",
  primaryContainer: "rgb(10, 74, 121)",
  onPrimaryContainer: "rgb(207, 228, 255)",
  error: "rgb(255, 180, 171)",
  onError: "rgb(105, 0, 5)",
  errorContainer: "rgb(147, 0, 10)",
  onErrorContainer: "rgb(255, 180, 171)",
  background: "rgb(26, 28, 30)",
  onBackground: "rgb(226, 226, 229)",
  surface: "rgb(66, 71, 78)",
  onSurface: "rgb(194, 199, 207)",
  outline: "rgb(140, 145, 153)",
  elevation: {
    level0: "transparent",
    level1: "rgb(32, 37, 41)",
  },
  surfaceDisabled: "rgba(226, 226, 229, 0.12)",
  onSurfaceDisabled: "rgba(226, 226, 229, 0.38)",
};

export default { light, dark };
