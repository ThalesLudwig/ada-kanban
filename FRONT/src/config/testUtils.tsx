import { Provider } from "react-redux";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import { PersistGate } from "redux-persist/integration/react";

import ThemeProvider from "../components/ThemeProvider/ThemeProvider";
import store, { persistor } from "./store.ts";

type Props = { children: React.ReactNode };

export const TestProvider: React.FC<Props> = ({ children }) => {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistor}>
        <ThemeProvider>
          <DndProvider backend={HTML5Backend}>{children}</DndProvider>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  );
};
