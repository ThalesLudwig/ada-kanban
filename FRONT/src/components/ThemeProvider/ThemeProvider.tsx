import "react-toastify/dist/ReactToastify.css";

import { ReactNode } from "react";
import { useSelector } from "react-redux";
import { ToastContainer } from "react-toastify";
import {
  ThemeProvider as StyledProvider,
  createGlobalStyle,
} from "styled-components";

import theme from "../../config/theme";
import { RootState } from "../../config/store";

type ConnectedThemeType = { children: ReactNode };

const GlobalStyle = createGlobalStyle`
  body {
    background: ${({ theme }) => theme.background};
    font-family: Poppins, Segoe UI, Roboto, Helvetica Neue, sans-serif;
    color-scheme: light dark;
    font-synthesis: none;
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    padding: 0;
    margin: 0;
  }
`;

const ThemeProvider = ({ children }: ConnectedThemeType) => {
  const { isDarkMode } = useSelector((store: RootState) => store.theme);

  return (
    <StyledProvider theme={isDarkMode ? theme.dark : theme.light}>
      <GlobalStyle />
      <ToastContainer
        closeOnClick
        pauseOnFocusLoss={false}
        pauseOnHover={false}
        theme={isDarkMode ? "dark" : "light"}
      />
      {children}
    </StyledProvider>
  );
};

export default ThemeProvider;
