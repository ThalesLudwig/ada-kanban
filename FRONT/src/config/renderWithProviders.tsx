import { render } from "@testing-library/react";
import { TestProvider } from "./testUtils";

export const renderWithProviders = (children: React.ReactNode) => {
  return render(<TestProvider>{children}</TestProvider>);
};
