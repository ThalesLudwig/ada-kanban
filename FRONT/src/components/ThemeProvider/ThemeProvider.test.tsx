import { afterEach, describe, expect, test, vi } from "vitest";
import { cleanup } from "@testing-library/react";

import ThemeProvider from "./ThemeProvider";
import { renderWithProviders } from "../../config/renderWithProviders";

describe("ThemeProvider Component", () => {
  afterEach(() => {
    vi.clearAllMocks();
    cleanup();
  });

    test("should render correctly", () => {
        const { container, findByText } = renderWithProviders(<ThemeProvider>foobar</ThemeProvider>);
        expect(container).toMatchSnapshot();
        expect(findByText("foobar")).toBeTruthy();
    });
});
