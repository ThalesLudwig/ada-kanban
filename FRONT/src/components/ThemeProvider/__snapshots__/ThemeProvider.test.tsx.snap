// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeProvider Component > should render correctly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="Toastify"
  />
  foobar
</div>
`;
